### ISSUE TITLE
PHPExcel is deprecated

### ISSUE THREAD

**Issue by gopi-ar** [portphp/portphp#87]
From the note on the [repo](https://github.com/PHPOffice/PHPExcel), they recommend shifting to https://github.com/PHPOffice/PhpSpreadsheet or another library.

**Comment by infomaniac50** [comment]
I made a conversion here https://github.com/infomaniac50/portphp-spreadsheet. I made the conversion in about an hour. No guarantees it will work, but all of the tests pass.

Follow https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository to use my fork.

The package was renamed to portphp/spreadsheet and the Excel namespace was changed to Spreadsheet.
In your PHP code import it like this.
```php
use Port/Spreadsheet/ExcelReader;
```

In your composer.json add these objects.
```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/infomaniac50/portphp-spreadsheet"
        }
    ],
    "require": {
        "portphp/spreadsheet": "dev-master"
    }
}
```

**Comment by gopi-ar** [comment]
If you can get that merged into this repo, it'll help the rest of the community. :-)

**Comment by infomaniac50** [comment]
PhpSpreadsheet has breaking changes in it, so that's why I changed the package name. I'm not sure if it will perform the same as the original. I haven't had a chance to test it in an application yet. Once it's in production I'll know if there are any more kinks to iron out.

The PHPOffice devs deprecated PHPExcel and did a major rewrite of it to make PHPSpreadsheet. I'm wondering if the PortPHP devs might want to do the same. If they are fine with merging it directly into this repository then I'll make a pull request.

**Comment by infomaniac50** [comment]
@ddeboer How much of https://github.com/infomaniac50/portphp-spreadsheet would you like to integrate?

**Comment by sagikazarmark** [comment]
I think we should migrate to the new package as well and maybe even rename the package to spreadsheet as well. We are free to do that in a new major version.

I suggest opening a PR to the current repository, once it lands in the master we can decide on the package naming and tag a 2.0 version.

**Comment by infomaniac50** [comment]
OK. I'll be back in week after I have used it in an app I'm working on. I'll clean up any logic errors and make a pull request.

**Comment by infomaniac50** [comment]
Here's the list of Manual Changes in case anyone is interested. https://phpspreadsheet.readthedocs.io/en/develop/topics/migration-from-PHPExcel/#manual-changes

The column offset was the only big change for PortPHP. Column 0 maps to Z and 1 maps to A.

**Comment by infomaniac50** [comment]
Who is the authority for @portphp?

**Comment by infomaniac50** [comment]
I decided to revert all of my name changes. https://github.com/infomaniac50/portphp-spreadsheet/tree/clean contains all of the latest updates now.

https://github.com/portphp/csv/blob/master/src/CsvReader.php is pretty nice. It should be ported to ExcelReader at some point. Specifically the column header code in current(). 

TBH I'm not sure if IOFactory::createReaderForFile will return anything for a CSV file.

**Comment by ddeboer** [comment]
Hey @infomaniac50, thanks for your effort!

I created a new (empty) repo at https://github.com/portphp/spreadsheet. Can you please submit a PR to that repo with your changes?

After that’s been done, we should also update the [docs](https://github.com/portphp/portphp/tree/master/docs).

**Comment by lsmith77** [comment]
@ddeboer BTW the new lib seems to do the trick!

**Comment by ddeboer** [comment]
Cool, good to hear!
